File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        result = list()

        for i in range(numRows):
            if not result:
                result.append([1])
            else:
                result.append(list())
                result[-1].append(1)
                for j in range(i - 1):
                    result[-1].append(result[-2][j] + result[-2][j + 1])
                result[-1].append(1)

        return result

File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) {
        vector<vector<int>> result;

        for (int i = 0; i < numRows; i++) {
            vector<int> temp;
            temp.push_back(1);

            if (!result.empty()) {
                for (int j = 0; j < i - 1; j++) {
                    int target = result.back()[j] + result.back()[j + 1];
                    temp.push_back(target);
                }
                temp.push_back(1);
            }

            result.push_back(temp);
        }

        return result;
    }
};